### ISSUE TITLE
No long auth login just: INF email= name=" " success=true (Re-Open Ref #246)

### ISSUE THREAD

**Issue by MyFriendlyUserName** [majd/ipatool#307]
### What happened?

Referencing #246 as reference was auto-closed and issue is still there in version 2.1.4 (ARM64):
```
$ ipatool-2.1.4-macos-arm64 auth login -e <apple-id-email>
9:24AM INF enter password:
9:24AM INF email= name=" " success=true
```
When trying to search something:
```
$ ipatool-2.1.4-macos-arm64 search test
9:23AM ERR error="country code is invalid: country code mapping for store front () was not found" success=false
```
Country Code should be AT.

Thanks for your help!

### Version

2.1.4

Please let me know, how I can help you to fix this issue.

**Comment by ionuttofan** [comment]
You could manually add the store front code in your keychain
{"name":"","password":"***","storefront":"code"}
find your store front code here
https://github.com/majd/ipatool/blob/63ee6fc6a42a89d51c9caac632cefd65218825ab/pkg/appstore/storefront.go

**Comment by ionuttofan** [comment]
look for keychain acces, then open keychain acces, after this on login screen, look for ipatool-auth.service. Double click on that and on the end of the page, you'll see "show password". Check that box and you can edit everything

**Comment by BreakfastSerial** [comment]
I was facing the same issue and resolved it on Linux (Kali) by modifying the "Keystore" entry as follows:

- Log in as described 
- Identify where ipatool stores the data (gnome-keyring for me) and find a method to modify these entries (I've installed seahorse)
- The "password" for the keyring entry of ipatool will look like: `{"Key":"account","Data":"ey[...]","Label":"","Description":"","KeychainNotTrustApplication":false,"KeychainNotSynchronizable":false}`
- The relevant bit here, is the base64 encoded value of `"Data":"ey[...]"`
- Base64 decode the value (e.g. using https://www.base64decode.org/)
- Add "storefront":"1234" (take the appropriate value from https://github.com/majd/ipatool/blob/8a20cfcb1defae0581156e8ff90f2e6105b9d65b/pkg/appstore/store_front.go), the resulting value should look like: `{"name":" ","password":"hunter2","storefront":"143456"}` (obviously, adjust the storefront ID and keep your own password)
- Base64 encode the value (e.g. using https://www.base64encode.org/)
- Overwrite the "Data" value in the password entry of the keyring/store/chain you've identified before.
- Test `./ipatool-2.1.4-linux-amd64 search "test"` should work now

**Comment by eduedp** [comment]
Where is the key stored in windows ?
There is a .ipatool folder in user path, but content is encoded in some weird structure I can't figure it out.

**Comment by Corndoglover69** [comment]
Cann someone tell me what a keychain is???

**Comment by FlashTang** [comment]
> keychain

If you are using macOS , go to Lounchpad ->  Other -> Keychain Access

**Comment by Corndoglover69** [comment]
> > keychain
> 
> If you are using macOS , go to Lounchpad -> Other -> Keychain Access

using windows

**Comment by majd** [comment]
The issue should be fixed in #316.
